node-export-server/server.js

```diff
@@ -224,10 +281,136 @@ exports.validateBbox = function(bbox) {
     return null;
 }
 
+/**
+ * updates job with hash's status
+ * We do this in a couple places so figured a function would reduce writing smame things twice.
+ */
+function jobComplete(hash) {
+    jobs[hash].status = 'complete';
+    console.log(jobs[hash].id + ' complete');
+}
+
+/**
+ * Creates a zip archive of outFile.
+ */
+function zipOutput(hash,output,outFile,outDir,outZip,isFile,format,cb) {
+    var stream = fs.createWriteStream(outZip);
+    stream.on('close', function() {
+        jobComplete(hash);
+        if (cb) cb(); // optional callback function. used now in the multipolygon case so we can unlink each of the input files once the zip is generated.
+    });
+
+    var zip = archiver('zip');
+    zip.on('error', function(err) {
+        throw err;
+    })
+    zip.pipe(stream);
+
+    if (isFile) {
+        zip.file(outFile, { name: output + config.formats[format]});
+    } else {
+        //Don't include file extension for shapefile
+        var ext = (format === 'Shapefile') ? '' : config.formats[format];
+        zip.directory(outDir, output + ext);
+    }
+    zip.finalize();
+}
+
+/**
+ * Builds the hootenanny command(s) from parts of provided request.
+ */
+function buildCommand(paramschema, queryOverrideTags, querybbox, querypoly, isFile, input, outDir, outFile, doCrop, ignoreSourceIds, mutlipleInputs) {
```

Comment:
this gets used
1. when a request specifies just a bbox or singleton polygon
2. when we are exporting the polygons that make up a multipolygon in the case the poly query parameter represents a multipolygon
3. when we take the outputs of exported mutlipolygon polygons and merge them into a single file.

The last two arguments are new. doCrop, when true and when we have a polygon or bbox, will crop the output of the hoot convert. ignoreSourceIds handles the negativeId collision that happens when exporting mutlipolygons.